core/src/main/scala/io/prediction/core/CleanedDataSource.scala

```diff
@@ -116,11 +121,52 @@ trait CleanedDataSource {
   def cleanedPEvents(sc: SparkContext): RDD[Event] = {
     val stage1 = getCleanedPEvents(sc)
     val result = cleanPEvents(sc, stage1)
+    val originalEvents = PEventStore.find(appName)(sc)
     val (appId, channelId) = Common.appNameToId(appName, None)
-    PEventStore.wipe(result, appId, channelId)(sc)
+    wipe(result.collect.toSet, originalEvents.collect.toSet, appId, channelId)
+    //PEventStore.wipe(result, appId, channelId)(sc)
     result
```

Comment:
yes, just a convenience . agree there is a command/query separation problem going on here. there is already another method with that name though. Was going for minimal changes to Max's work here, can refactor separately